tests/unit/test_charm.py

```diff
@@ -45,8 +48,14 @@ def test_config_changed():
         config={"server-port": 8080},
         leader=True,
     )
-    ctx.run(ctx.on.config_changed(), state_in)
-    assert "--port=8080" in container.layers["fastapi_demo"].services["fastapi-service"].command
+    state_out = ctx.run(ctx.on.config_changed(), state_in)
+    assert (
+        "--port=8080"
+        in state_out.get_container(container.name)
+        .layers["fastapi_demo"]
+        .services["fastapi-service"]
+        .command
```

Comment:
I don't love the layout here - I assume it was auto formatted. Probably pulling out the service in a separate line would read better, but I think it's probably fine to leave as-is.